added ent to ce downgrade doc

aahel opened the pull request:
### Description

<!-- Please describe why you're making this change, in plain English. -->

### Testing & Reproduction steps

<!--

* In the case of bugs, describe how to replicate
* If any manual tests were done, document the steps and the conditions to replicate
* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding

-->

### Links

<!--

Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.

-->

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

mkeeler commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,110 @@
+- Filter out tenanted data from the request
+
+  For example :-
+  - Acl Token Batch Set requests -  acl  tokens with non default namespace or partiton are filtered out.
+  - Acl Policy Batch Set requests -  acl  policies with non default namespace or partiton are filtered out.
```
```suggestion
  - Intention Sources - Intention sources targeting non default namespaces or partitions are filtered out of the config entry.
  - Exported Services - Exports of services within non default namespaces or partitions are filtered out of the config entry.
```

I think these are probably a better examples as they are directly applicable to user data. The Batch requests for ACLs is an internal detail that most users wouldn't know about.

mkeeler commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,110 @@
+  - Acl Token Batch Set requests -  acl  tokens with non default namespace or partiton are filtered out.
+  - Acl Policy Batch Set requests -  acl  policies with non default namespace or partiton are filtered out.
+
+- Panic and stop the downgrade
```
It is probably worth an extra sentence here to note why we have to panic. Something to the effect of:

> Config Entries regarding traffic routing cannot be safely filtered because Consul  cannot know whether the resulting filtered config entries send traffic towards  services which can handle that traffic. In these situations, you are required to first remove references to services within non default namespaces or partitions from within those config entries. When encountering these config entries Consul CE in the downgrade mode will panic in order to not cause harm to existing service mesh routing.

mkeeler reviewed (commented):
I left a few suggestions but overall I think this level of detail is probably fine.

This PR will require approval from someone on the docs team before merging as well.

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+## What happens during the downgrade
+
+During the downgrade as the raft replication logs start flowing into the ce server we handle them in one of the following ways :- 
```
```suggestion
During the downgrade as the raft replication logs start flowing into the ce server we handle them in one of the following ways:
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+- Drop the request
+
+   For example :-
```
```suggestion
   For example :
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+- Filter out tenanted data from the request
+
+  For example :-
```
```suggestion
  For example :
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+  Config Entries regarding traffic routing cannot be safely filtered because Consul cannot know whether the resulting filtered config entries send traffic towards services which can handle that traffic. In these situations, you are required to first remove references to services within non default namespaces or partitions from within those config entries. When encountering these config entries Co ...
+  
+  For example :-
```
```suggestion
  For example :
```

david-yu commented on website/data/docs-nav-data.json:
```diff
@@ -1821,6 +1821,10 @@
         "title": "Admin Partitions",
         "path": "enterprise/admin-partitions"
       },
+      {
```
Could we move this to the bottom of the Enterprise section? It should be the last entry there.

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+## Perform the Downgrade
+
+- Restart or redeploy all clients with a CE version of the binary
```
```suggestion
1. Restart or redeploy all clients with a CE version of the binary
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+- Restart or redeploy all clients with a CE version of the binary
+
+- Update server binaries to use the CE version
```
```suggestion
2. Update server binaries to use the CE version
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+- Update server binaries to use the CE version
+
+- Restart or redeploy the servers one at a time with an env var named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` set to `true` on the servers.
```
```suggestion
3. Restart or redeploy the servers one at a time with an env var named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` set to `true` on the servers.
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+**3.** Remove references to tenanted services from service-resolver, service-splitter and service-router config entries.
+
+
+## Perform the Downgrade
```
@aahel Could we have actual CLI commands here? For example, for restart all clients how does one know how to restart them? Do you need to kill -9 or is there another command to restart clients?

aahel replied:
@david-yu i updated the downgrade steps a bit. referred to steps in the upgrade doc as the steps are similar

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+3. Validate that all servers are on CE and healthy using `consul members` command
+
+## What happens during the downgrade
+
+During the downgrade as the raft replication logs start flowing into the ce server we handle them in one of the following ways:
+
+- Drop the request
+
+   For example :
+   - Register requests - register requests having non default namespace, having services or health checks in non default namespace or partition are dropped.
+
+   - Peering Write Requests - peering write requests are dropped if the local partiton connecting to the peer is non default. 
+
+- Filter out tenanted data from the request
+
+  For example :
+  - Intention Sources - Intention sources targeting non default namespaces or partitions are filtered out of the config entry.
+  - Exported Services - Exports of services within non default namespaces or partitions are filtered out of the config entry.
+
+- Panic and stop the downgrade
+
+  Config Entries regarding traffic routing cannot be safely filtered because Consul cannot know whether the resulting filtered config entries send traffic towards services which can handle that traffic. In these situations, you are required to first remove references to services within non default namespaces or partitions from within those config entries. When encountering these config entries Co ...
+  
+  For example :
+  - Service Splitter,Service Resolver,Service Router Config Entry requests - if these config entries have references to tenanted services it will cause a panic.
```
```suggestion
## Downgrade process details

During the downgrade process, the Consul CE server handles the raft replication logs in one of the following ways:

- Drops the request. 
- Filters out data from requests sent from non-default namespaces and partitions
- Panics and stops the downgrade 

The server may drop the following types of requests: 

- Registration requests in non-default namespace and services or health checks in non-default namespaces or partitions.

- Write requests to peered clusters if the local partiton connecting to the peer is non-default. 

The server may filter out data from requests to non-default namespaces and partitions in the following cases:

- Intention sources that target non-default namespaces or partitions are filtered out of the configuration entry.
- Exports of services within non-default namespaces or partitions are filtered out of the configuration entry.

The server may panic and stop the downgrade when configuration entries that route traffic cannot be safely filtered. This is because Consul cannot know whether the resulting filtered configuration entries send traffic to services that can handle the traffic. Consul CE panics in order to prevent harm to existing service mesh routes.

In these situations, you must first remove references to services within non-default namespaces or partitions from those configuration entries. 
  
The server may panic in the following cases:

- Service splitter, service resolver, and service router configuration entry requests that have references to services located in non-default namespaces and partitions cause the server to panic.
```
Cleaned up some grammar and style guide inconsistencies. 
I think this section should appear immediately after the Introduction section. It is background information to help readers understand if they want to perform the procedure described in the topic. 
The term "tenanted" is sort of difficult to understand right away, so I changed it to "non-default namespaces and partitions". Is that what we mean? It's a little longer, but does not require as much interpretation.

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+## Introduction
+
+This document describes some best practices that you should follow when
+downgrading from Consul Enterprise to Consul CE.
```
```suggestion
This document describes how to downgrade from Consul Enterprise to Consul community edition (CE).
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+This document describes some best practices that you should follow when
+downgrading from Consul Enterprise to Consul CE.
+
+## Download the CE Version
```
```suggestion
## Download the CE binary version
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+<Tabs>
+<Tab heading="Binary">
+
+All current and past versions of the CE and Enterprise releases are
+available here:
+
+- https://releases.hashicorp.com/consul
```
```suggestion
All current and past versions of the CE and Enterprise releases are
available on the [HashiCorp releases page](https://releases.hashicorp.com/consul)
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+</Tab>
+<Tab heading="Docker">
+
+Docker containers are available at this location:
+
+- **CE:** https://hub.docker.com/r/hashicorp/consul
```
```suggestion
Docker containers are available on [Docker Hub](https://hub.docker.com/r/hashicorp/consul)
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+</Tabs>
+
+## Prepare for the Downgrade to CE
```
```suggestion
## Prepare for the downgrade to CE
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,112 @@
+consul snapshot save backup.snap
+```
+
+You can inspect the snapshot to ensure if was successful with:
````
```suggestion
1. Inspect the snapshot to ensure that Consul successfully captured it using the following command:
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,112 @@
+Version      1
+```
+
+This will ensure you have a safe fallback option in case something goes wrong. Store
+this snapshot somewhere safe. More documentation on snapshot usage is available here:
+
+- [consul.io/commands/snapshot](/consul/commands/snapshot)
+- [Backup Consul Data and State tutorial](/consul/tutorials/production-deploy/backup-and-restore)
````
```suggestion
    Inspecting the snapshot ensures that you have a safe fallback option in case something goes wrong. Store
this snapshot somewhere safe. Refer to the following documentation for additional information about creating and using snapshots:

    - [Consul Snapshot](/consul/commands/snapshot)
    - [Backup Consul Data and State tutorial](/consul/tutorials/production-deploy/backup-and-restore)
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+## Prepare for the Downgrade to CE
+
+**1.** Take a snapshot:
```
```suggestion
1. Take a snapshot:
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,112 @@
+
+**1.** Take a snapshot:
+
+```
+consul snapshot save backup.snap
+```
````
```suggestion
    ```
    consul snapshot save backup.snap
    ```
```
We have to indent everything between each list item to use the markdown correctly

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,112 @@
+
+You can inspect the snapshot to ensure if was successful with:
+
+```
+consul snapshot inspect backup.snap
+```
+
+Example output:
+
+```
+ID           2-1182-1542056499724
+Size         4115
+Index        1182
+Term         2
+Version      1
+```
````
```suggestion
    ```
    consul snapshot inspect backup.snap
    ```

    Example output:

    ```
    ID           2-1182-1542056499724
    Size         4115
    Index        1182
    Term         2
    Version      1
    ```
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,112 @@
+- [consul.io/commands/snapshot](/consul/commands/snapshot)
+- [Backup Consul Data and State tutorial](/consul/tutorials/production-deploy/backup-and-restore)
+
+**2.** Temporarily modify your Consul configuration so that its [log_level](/consul/docs/agent/config/cli-flags#_log_level)
+is set to `debug`. After doing this, issue the following command on your servers to
+reload the configuration:
+
+```
+consul reload
+```
+
+This change will give you more information to work with in the event something goes wrong.
+
+
+**3.** Remove references to tenanted services from service-resolver, service-splitter and service-router config entries.
+
````
```suggestion
1. Temporarily modify your Consul configuration so that its [log_level](/consul/docs/agent/config/cli-flags#_log_level)
is set to `debug`. After doing this, issue the following command on your servers to
reload the configuration:

    ```
    consul reload
    ```

   This change gives you more information to work with in the event something goes wrong.


1. Remove references to services in non-default namespaces and partitions from service resolver, service splitter, and service router configuration entries.

```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+**3.** Remove references to tenanted services from service-resolver, service-splitter and service-router config entries.
+
+
+## Perform the Downgrade
```
```suggestion
## Perform the downgrade
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,112 @@
+
+3. Restart or redeploy the servers one at a time with an env var named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` set to `true` on the servers.
+
+3. Validate that all servers are on CE and healthy using `consul members` command
```
```suggestion
1. Restart or redeploy all clients with a CE version of the binary.

1. Update server binaries to use the CE version.

1. Restart or redeploy the servers one at a time with an environment variable named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` set to `true` on the servers.

1. Validate that all servers are on CE and healthy using `consul members` command.
```

aahel replied:
have updated the content of this part.  so not adding these changes

trujillo-adam commented on website/data/docs-nav-data.json:
```diff
@@ -1890,6 +1890,10 @@
         ]
+      },
+      {
+        "title": "Enterprise to CE Downgrade",
```
```suggestion
        "title": "Enterprise to CE downgrade",
```

trujillo-adam reviewed (commented):
There are some grammar and style guide things we need to clean up, but the content is overall really clear.

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,173 @@
+  Guide to downgrade from consul enterprise to consul ce
+---
+
+## Introduction
```
```suggestion
# Downgrade from Consul Enterprise to the community edition
```
We need to use a meaningful title in the H1 for searchability.

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,173 @@
+---
+layout: docs
+page_title: Enterprise to CE downgrade Process
```
```suggestion
page_title: Downgrade from Consul Enterprise to the community edition
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,173 @@
+layout: docs
+page_title: Enterprise to CE downgrade Process
+description: >-
+  Guide to downgrade from consul enterprise to consul ce
```
```suggestion
  Learn how to downgrade your installation of Consul Enterprise to the free Consul community edition (CE)
```
We have a template for writing metadescriptions.

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,173 @@
+## Prepare for the downgrade to CE
+
+1. Take a snapshot:
+
+    ```
+    consul snapshot save backup.snap
+    ```
+
````
```suggestion

   ```
   consul snapshot save backup.snap
   ```

```
Apologies - I added one too many spaces in my last suggestion, which messed up the formatting. For future reference, it's 3 spaces, not 4. This looks weird in the GH UI, but it should fix the rendering. I'll do this in all the other places I messed it up.

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,173 @@
+    ```
+
+1. Inspect the snapshot to ensure that Consul successfully captured it using the following command:
+
+    ```
+    consul snapshot inspect backup.snap
+    ```
+
+    Example output:
+
+    ```
+    ID           2-1182-1542056499724
+    Size         4115
+    Index        1182
+    Term         2
+    Version      1
+    ```
+
+    Inspecting the snapshot ensures that you have a safe fallback option in case something goes wrong. Store
````
```suggestion

   ```
   consul snapshot inspect backup.snap
   ```

   Example output:

   ```
   ID           2-1182-1542056499724
   Size         4115
   Index        1182
   Term         2
   Version      1
   ```

   Inspecting the snapshot ensures that you have a safe fallback option in case something goes wrong. Store
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,173 @@
+1. Temporarily modify your Consul configuration so that its [log_level](/consul/docs/agent/config/cli-flags#_log_level)
+is set to `debug`. After doing this, issue the following command on your servers to
+reload the configuration:
+
+    ```
+    consul reload
+    ```
+
+   This change gives you more information to work with in the event something goes wrong.
+
+
````
```suggestion

   ```
   consul reload
   ```

   This change gives you more information to work with in the event something goes wrong.

```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,173 @@
+this should avoid an unexpected leadership election due to loss of quorum.
+
+**5.** Double-check that all servers are showing up in the cluster as expected and are on
+the correct version by issuing:
+
+```
+consul members
+```
+
+You should receive output similar to this:
+
+```
+Node       Address         Status  Type    Build  Protocol  DC
+dc1-node1  10.11.0.2:8301  alive   server  1.8.3  2         dc1
+dc1-node2  10.11.0.3:8301  alive   server  1.8.3  2         dc1
+dc1-node3  10.11.0.4:8301  alive   server  1.8.3  2         dc1
+```
+
+Also double-check the raft state to make sure there is a leader and sufficient voters:
+
+```
+consul operator raft list-peers
+```
+
+You should receive output similar to this:
+
+```
+Node       ID                                    Address         State     Voter  RaftProtocol
+dc1-node1  ae15858f-7f5f-4dcb-b7d5-710fdcdd2745  10.11.0.2:8300  leader    true   3
+dc1-node2  20e6be1b-f1cb-4aab-929f-f7d2d43d9a96  10.11.0.3:8300  follower  true   3
+dc1-node3  658c343b-8769-431f-a71a-236f9dbb17b3  10.11.0.4:8300  follower  true   3
+```
+
+**6.** Set your `log_level` back to its original value and issue the following command
+on your servers to reload the configuration:
+
+```
+consul reload
+```
+
````
```suggestion
1. Issue the following command to discover which server is currently the leader:

   ```
   consul operator raft list-peers
   ```

   Consul lists the raft peers in a format similar to the following output. The exact format and content may differ based on version:

   ```
   Node       ID                                    Address         State     Voter  RaftProtocol
   dc1-node1  ae15858f-7f5f-4dcb-b7d5-710fdcdd2745  10.11.0.2:8300  leader    true   3
   dc1-node2  20e6be1b-f1cb-4aab-929f-f7d2d43d9a96  10.11.0.3:8300  follower  true   3
   dc1-node3  658c343b-8769-431f-a71a-236f9dbb17b3  10.11.0.4:8300  follower  true   3
   ```
   Take note of which agent is the leader.

1. Restart or redeploy all clients with a CE version of the binary. You can use a service management system, such as `systemd` or `upstart`, to restart the Consul service. If you are not using a service management system, you must restart the  agent manually.

1. Update server binaries to use the CE version.

1. Complete the following steps in order on the server agents, leaving the leader
agent for last. 
      1. Set and environment variable named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` to `true` on the servers. 
      1. Then use a service management system, such as `systemd` or `upstart`, to restart the Consul service. If you are not using a service management system, you must restart the agent manually.

1. Issue the following command to validate that the agent has rejoined the cluster and is in sync with the leader:

   ```
   consul info
   ```

   The `commit_index` and `last_log_index` fields should have the same value. If done properly,
this should avoid an unexpected leadership election due to loss of quorum.

1. Double-check that all servers are showing up in the cluster as expected and are on
the correct version by issuing the following command:

   ```
   consul members
   ```

   Consul lists the members:

   ```
   Node       Address         Status  Type    Build  Protocol  DC
   dc1-node1  10.11.0.2:8301  alive   server  1.8.3  2         dc1
   dc1-node2  10.11.0.3:8301  alive   server  1.8.3  2         dc1
   dc1-node3  10.11.0.4:8301  alive   server  1.8.3  2         dc1
   ```

1. Double-check the raft state to make sure there is a leader and sufficient voters:

   ```
   consul operator raft list-peers
   ```

   Consul lists the rafter peers in a format similar to the following output:

   ```
   Node       ID                                    Address         State     Voter  RaftProtocol
   dc1-node1  ae15858f-7f5f-4dcb-b7d5-710fdcdd2745  10.11.0.2:8300  leader    true   3
   dc1-node2  20e6be1b-f1cb-4aab-929f-f7d2d43d9a96  10.11.0.3:8300  follower  true   3
   dc1-node3  658c343b-8769-431f-a71a-236f9dbb17b3  10.11.0.4:8300  follower  true   3
   ```

1. Set your `log_level` back to its original value and issue the following command
on your servers to reload the configuration:

   ```
   consul reload
   ```

```
This suggestion fixes the formatting to use MD lists and addresses style guide inconsistencies.

aahel replied:
i think we should address formatting issues  at the end once the content is fully approved. so that we fix it at once

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,173 @@
+consul reload
+```
+
+## Downgrade process details
````
```suggestion
## Downgrade process details
```
I still believe that this should be near the top so that users understand what is going to happen prior to complete this process.

aahel replied:
do you recmmend making this the first section the doc. ie before downlaod consul section

trujillo-adam replied:
Normally, a usage page follows this format:

1. **Description**: This is the very first sentence or two under page heading that tells the reader what the page is about. 
2. **Introduction**: This section contains background information or provides an overview or preview of procedures the page is about to describe. _This is where I think this information should go, along with any additional justification for downgrading from Enterprise to CE._
3. **Requirements**: Requirements or prerequisites
4. **Procedures**: The main part of the topic described in steps.

aahel replied:
should i rename `Downgrade from Consul Enterprise to the community edition` section to introduction ? 
and put all the downgrade details inside that ?
or maybe create a downgrade section after that ?
can you add a suggestion where you would put it in the current doc

aahel replied:
this doc is mostly based on https://developer.hashicorp.com/consul/docs/upgrading/instructions/general-process

trujillo-adam reviewed (commented):
I apologize, but some of the formatting suggestions I made in my last review introduced formatting errors. I ran a copy locally to test my suggestions this time. We also need to follow the style guide more consistently.

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+
+## Prepare for the downgrade to CE
+
+1. Take a snapshot:
```
We should either link to how to take a snapshot or directly provide an example for taking a snapshot here.

aahel replied:
cli command is there. i have reused this step from here https://developer.hashicorp.com/consul/docs/upgrading/instructions/general-process#prepare-for-the-upgrade

aahel replied:
the snapshot commands got removed by mistake in one of the previous suggested comments.
let me know if its fine know

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+1. Inspect the snapshot to ensure that Consul successfully captured it using the following command:
+this snapshot somewhere safe. Refer to the following documentation for additional information about creating and using snapshots:
+
+    - [Consul Snapshot](/consul/commands/snapshot)
```
These bullets seem to not be formatted correctly if you look at the preview.

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+
+**3.** Update server binaries to use the CE version
+
+**4.** The following steps must be done in order on the server agents, leaving the leader
```
Could you we insert some CLI examples here? For example:

```
export CONSUL_ENTERPRISE_DOWNGRADE_TO_CE= true
``

Ranjandas replied:
I think it is better to show an example that sets this environment variable under systemd, as most of our users running on VMs would run under systemd. There is a chance that without this, users would try to simply set the environment variable in the shell and restart the service which would not work.

david-yu replied:
We should provide an example using systemd. An example is better than no example.

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+<Tabs>
+<Tab heading="Binary">
+
+All current and past versions of the CE and Enterprise releases are
```
It would be good to instead provide a curl command that downloads a version of Consul for example. 


```
export VERSION=1.17.0
curl https://releases.hashicorp.com/consul/1.17.0/consul_${VERSION}_linux_amd64.zip
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+available on the [HashiCorp releases page](https://releases.hashicorp.com/consul)
+
+</Tab>
+<Tab heading="Docker">
```
I believe in this case we should just call this "Kubernetes" instead. Also point to the helm value where the version is changed.

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,144 @@
+
+**1.** Issue the following command to discover which server is currently the leader:
+
+```
````
```suggestion
```shell-session
```

I would add shell-session to all CLI console commands

aahel replied:
have used ```shell.

Ranjandas commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+
+# Downgrade from Consul Enterprise to the community edition
+
+This document describes how to downgrade from Consul Enterprise to Consul community edition (CE).
```
We must call out from which version this downgrade method is supported.

Ranjandas commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+
+## Download the CE binary version
+
+First, download the binary for the CE version you want.
```
Just a thought; instead of specifically instructing to download the version, I think we can follow our standard upgrade guide format and say _install the required CE version_; and the installation specifics should be left to the user.

ref: https://developer.hashicorp.com/consul/docs/upgrading#standard-upgrades

aahel replied:
i had followed this doc where download is mentioned. https://developer.hashicorp.com/consul/docs/upgrading/instructions/general-process

do you want the entire download section to be removed ?

Ranjandas commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,144 @@
+1. Temporarily modify your Consul configuration so that its [log_level](/consul/docs/agent/config/cli-flags#_log_level)
+is set to `debug`. After doing this, issue the following command on your servers to
+reload the configuration:
+1. Remove references to services in non-default namespaces and partitions from service resolver, service splitter, and service router configuration entries.
```
I think this step would need more details about what must be done. Some of the immediate questions I have are:

* Should the config-entries in non-default namespaces/partitions be deleted?
* Should the non-default namespaces be deleted fully? 
* Or does this only apply to config entries in the default namespace and partition?

aahel replied:
the target services(of redirects,failovers,splits,routes) belonging to non default tenancies should be removed

aahel replied:
should i specify which parts of the config entry you need to remove (like failovers and redirects for service resolver) for non default tenancies ?

aahel replied:
have detailed the individual entries more.

Ranjandas commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,144 @@
+```
+Take note of which agent is the leader.
+
+**2.** Restart or redeploy all clients with a CE version of the binary. You can use a service management system (e.g., systemd, upstart, etc.) to restart the Consul service. If you are not using a service management system, you must restart the  agent manually.
````
I think it is better to separate the server and client downgrades. 

For eg: in this section, while the first step is finding the current leader, the second step jumps into downgrading clients which is not related to step 1.

Ranjandas commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,144 @@
+
+```
+Node       Address         Status  Type    Build  Protocol  DC
+dc1-node1  10.11.0.2:8301  alive   server  1.8.3  2         dc1
````
It is better we fix the build number in the example output, as using an older version would make the reader think that this functionality is supported in older versions as well.

aahel replied:
have modified the build to 1.18 since we are releasing this feature in 1.18

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,214 @@
+## Prepare for the downgrade to CE
+
+1. Take a snapshot.
+```shell
````
```suggestion
  ```shell
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,214 @@
+  ```
+2. You can inspect the snapshot to ensure if was successful with:
+
+```shell
+    consul snapshot inspect backup.snap
+```
````
```suggestion
  ```shell
   consul snapshot inspect backup.snap
  ```
```

david-yu replied:
```suggestion
  ```shell
   consul snapshot inspect backup.snap
  ```
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,214 @@
+```
+
+Example output:
+
+```shell
+ID           2-1182-1542056499724
+Size         4115
+Index        1182
+Term         2
+Version      1
+```
````
```suggestion

  ```shell
  ID           2-1182-1542056499724
  Size         4115
  Index        1182
  Term         2
  Version      1
  ```
```

david-yu commented:
I think this docs PR is pretty much almost ready to go but we need to have docs review formatting and likely style. Thanks @aahel !

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,214 @@
+
+This change will give you more information to work with in the event something goes wrong.
+
+4. Modify the following config entries if applicable :-
```
```suggestion
4. Modify the following config entries if applicable :
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,214 @@
+available on the [HashiCorp releases page](https://releases.hashicorp.com/consul)
+
+Example:-
+```shell
````
```suggestion
```shell-session
```

Should be shell-session not shell for all shell examples.

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,212 @@
+  ```
+2. You can inspect the snapshot to ensure if was successful with:
+
+  ```shell-session
+   consul snapshot inspect backup.snap
+
````
```suggestion
  ```shell-session
   consul snapshot inspect backup.snap
  ```
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,212 @@
+agent for last.
+ - An env var named `CONSUL_ENTERPRISE_DOWNGRADE_TO_CE` should be set to `true` on the servers. 
+
+   Following is an example to set the env var if you are using systemd :- 
```
```suggestion
   Following is an example to set the env var if you are using `systemd` :
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,212 @@
+
+4. Modify the following config entries if applicable :
+
+    service resolver: 
```
```suggestion
    [Service resolver](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-resolver): 
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,212 @@
+    1. remove  services configured as failovers in non default namespace or belonging to a sameness group. 
+    2. remove services configured as redirects belonging to non default namespaces or partiton.
+
+    service splitter: 
```
```suggestion
    [Service splitter](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-splitter): 
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,212 @@
+    service splitter: 
+    1. remove services configured as splits belonging to non default namespaces or partiton
+
+    service router:
```
```suggestion
    [Service router](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-router):
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -144,15 +144,14 @@ Take note of which agent is the leader.
-    consul info
-    ```
+      ```shell-session
+      consul info
````
```suggestion
      $ consul info
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,219 @@
+All current and past versions of the CE and Enterprise releases are
+available on the [HashiCorp releases page](https://releases.hashicorp.com/consul)
+
+Example:-
```
```suggestion
Example:
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,219 @@
+This will ensure you have a safe fallback option in case something goes wrong. Store
+this snapshot somewhere safe. More documentation on snapshot usage is available here:
+
+- [consul.io/commands/snapshot](/consul/commands/snapshot)
```
```suggestion
- [Consul snapshot](/consul/commands/snapshot)
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,219 @@
+on your servers to reload the configuration:
+
+```shell-session
+consul reload
````
```suggestion
$ consul reload
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,219 @@
+the correct version by issuing:
+
+```shell-session
+consul members
````
```suggestion
$ consul members
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,219 @@
+
+   Example :- 
+   ```shell-session
+   sudo systemctl restart consul
````
```suggestion
 $  sudo systemctl restart consul
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
````diff
@@ -0,0 +1,219 @@
+       reload the configuration:
+
+```
+consul reload
````
```suggestion
$ consul reload
```

david-yu commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,219 @@
+
+**3.** Modify the following config entries if applicable: 
+
+[Service resolver](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-resolver): 
+
+  1. Remove services configured as failovers in non default namespace or  belonging to a sameness group.
+
+  2. Remove services configured as redirects belonging to non default namespaces or partiton.
+
+[Service splitter](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-splitter):
+
+  1. Remove services configured as splits belonging to non default namespaces or partiton
+
+[Service router](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-router):
+
+  1. Remove services configured as destination belonging to non default namespaces or partiton
```
Is it possible to indent this section for formatting

aahel replied:
done

david-yu reviewed (approved):
Looks good to me, thank you for taking in all the suggestions to this doc. We should also wait for @trujillo-adam 's review as well.

trujillo-adam commented:
@aahel - I hope you don't mind, but it was just easier to push suggestions directly to this PR. I tweaked a few sentences and moved a few things around, but I don't think that I changed any of the meanings.

aahel commented:
> @aahel - I hope you don't mind, but it was just easier to push suggestions directly to this PR. I tweaked a few sentences and moved a few things around, but I don't think that I changed any of the meanings.

Thnx for this.Can you approve the pr if its ready to be merged

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,207 @@
+
+- Write requests to peered clusters if the local partition connecting to the peer is non-default.
+
+Following examples describe scenarios where the server may filter out data from requests:
```
```suggestion
The following examples describe scenarios where the server may filter out data from requests:
```

trujillo-adam commented on website/content/docs/enterprise/ent-to-ce-downgrades.mdx:
```diff
@@ -0,0 +1,207 @@
+
+## Perform the downgrade
+
+1. Restart or redeploy all Consul clients with a CE version of the binary. You can use a service management system, such as `systemd`, `upstart` to restart the Consul service. If you are not using a service management system, you must restart the agent manually. The following example uses `systemctl` to restart the Consul service:
```
```suggestion
1. Restart or redeploy all Consul clients with a CE version of the binary. You can use a service management system, such as `systemd` or `upstart`, to restart the Consul service. If you are not using a service management system, you must restart the agent manually. The following example uses `systemctl` to restart the Consul service:
```
